src/Core/src/Platform/Gtk/MauiDatePicker.cs

```diff
@@ -1,35 +1,443 @@
+		{
+			get
+			{
+				return _calendar.MaximumDate;
+			}
+
+			set
+			{
+				_calendar.MaximumDate = value;
+			}
+		}
+
+		protected virtual void OnButtonPressEvent(object o, ButtonPressEventArgs args)
+		{
+			CloseCalendar();
+		}
+
+		protected virtual void OnCalendarButtonPressEvent(object o, ButtonPressEventArgs args)
+		{
+			args.RetVal = true;
+		}
+
+		protected virtual void OnCalendarDaySelected(object? sender, EventArgs e)
+		{
+			OnDateTimeChanged?.Invoke(this, new DateEventArgs(SelectedDate));
+		}
+
+		protected virtual void OnCalendarDaySelectedDoubleClick(object? sender, EventArgs e)
+		{
+			OnDateTimeChanged?.Invoke(this, new DateEventArgs(SelectedDate));
+			CloseCalendar();
+		}
+
+		void CloseCalendar()
+		{
+			GrabHelper.RemoveGrab(this);
+			Destroy();
+		}
+
+		void NotifyDateChanged()
+		{
+			OnDateTimeChanged?.Invoke(this, new DateEventArgs(SelectedDate));
+		}
+
+		class RangeCalendar : Calendar
+		{
+			DateTime _minimumDate;
+			DateTime _maximumDate;
+
+			public RangeCalendar()
+			{
+				_minimumDate = new DateTime(1900, 1, 1);
+				_maximumDate = new DateTime(2199, 1, 1);
+			}
+
+			public DateTime MinimumDate
+			{
+				get
+				{
+					return _minimumDate;
+				}
+
+				set
+				{
+					if (MaximumDate < value)
+					{
+						throw new InvalidOperationException($"{nameof(MinimumDate)} must be lower than {nameof(MaximumDate)}");
+					}
+
+					_minimumDate = value;
+				}
+			}
+
+			public DateTime MaximumDate
+			{
+				get
+				{
+					return _maximumDate;
+				}
+
+				set
+				{
+					if (MinimumDate > value)
+					{
+						throw new InvalidOperationException($"{nameof(MaximumDate)} must be greater than {nameof(MinimumDate)}");
+					}
+
+					_maximumDate = value;
+				}
+			}
+
+			protected override void OnDaySelected()
+			{
+				if (Date < MinimumDate)
+				{
+					Date = MinimumDate;
+				}
+				else if (Date > MaximumDate)
+				{
+					Date = MaximumDate;
+				}
+			}
 		}
+	}
+
+	public class CustomComboBox : Gtk.HBox
+	{
+		private Gtk.Entry _entry;
+		private Gtk.Button _button;
+		private Gtk.Arrow _arrow;
+		private Color _color = Colors.Black;
+
+		public CustomComboBox()
+		{
+			_entry = new Gtk.Entry();
+			_entry.CanFocus = true;
+			_entry.IsEditable = true;
+			PackStart(_entry, true, true, 0);
+
+			_button = new Gtk.Button();
+			_button.WidthRequest = 30;
+			_button.CanFocus = true;
+			_arrow = new Gtk.Arrow(Gtk.ArrowType.Down, Gtk.ShadowType.EtchedOut);
+			_button.Add(_arrow);
+			PackEnd(_button, false, false, 0);
+		}
+
+		public Gtk.Entry Entry
+		{
+			get
+			{
+				return _entry;
+			}
+		}
+
+		public Gtk.Button PopupButton
+		{
+			get
+			{
+				return _button;
+			}
+		}
+
+		public Color Color
+		{
+			get { return _color; }
+			set
+			{
+				_color = value;
+				Entry.UpdateTextColor(_color);
+			}
+		}
+
+		public void SetBackgroundColor(Gdk.Color color)
+		{
+			var redColor = new Gdk.Color(255, 0, 0);
+			var blueColor = new Gdk.Color(0, 0, 255);
```

Comment:
@parhamsaremi is XamarinForms' DatePicker hardcoding these colors above too?

Reply:
They were doing something like this. so we created these colors and used them :thinking:  https://github.com/xamarin/Xamarin.Forms/blob/5.0.0/Xamarin.Forms.Platform.GTK/Controls/CustomComboBox.cs#L46

Reply:
why don't you use Maui's Red and Blue colors then? much more readable than 25500/00255